scm/src/gmtb_scm_output.F90

```diff
@@ -638,83 +828,104 @@ subroutine output_append(scm_state, physics)
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Radtend(i)%sfcfsw(1)%dnfxc
+     dummy_1D(i) = physics%Radtend%sfcfsw(i)%dnfxc
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="sw_dn_sfc_tot",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Radtend(i)%sfcfsw(1)%upfx0
+     dummy_1D(i) = physics%Radtend%sfcfsw(i)%upfx0
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="sw_up_sfc_clr",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Radtend(i)%sfcfsw(1)%dnfx0
+     dummy_1D(i) = physics%Radtend%sfcfsw(i)%dnfx0
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="sw_dn_sfc_clr",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Diag(i)%topflw(1)%upfxc
+     dummy_1D(i) = physics%Diag%topflw(i)%upfxc
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="lw_up_TOA_tot",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Diag(i)%topflw(1)%upfx0
+     dummy_1D(i) = physics%Diag%topflw(i)%upfx0
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="lw_up_TOA_clr",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Radtend(i)%sfcflw(1)%upfxc
+     dummy_1D(i) = physics%Radtend%sfcflw(i)%upfxc
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="lw_up_sfc_tot",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Radtend(i)%sfcflw(1)%dnfxc
+     dummy_1D(i) = physics%Radtend%sfcflw(i)%dnfxc
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="lw_dn_sfc_tot",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Radtend(i)%sfcflw(1)%upfx0
+     dummy_1D(i) = physics%Radtend%sfcflw(i)%upfx0
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="lw_up_sfc_clr",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
   !
   do i=1, scm_state%n_cols
-     dummy_1D(i) = physics%Radtend(i)%sfcflw(1)%upfx0
+     dummy_1D(i) = physics%Radtend%sfcflw(i)%upfx0
   end do
   CALL CHECK(NF90_INQ_VARID(NCID=ncid,NAME="lw_dn_sfc_clr",VARID=var_id))
   CALL CHECK(NF90_PUT_VAR(NCID=ncid,VARID=var_id,VALUES=dummy_1D,START=(/1,scm_state%itt_out /)))
-
+  
```

Comment:
You could use the time_avg logical arrays in the same way as it is done in fv3atm - if it is .true. for a specific slice of the array, simply divide by the integration time (same as you do for the 3d tendencies).

Reply:
OK, I'll look into that for the output rewrite.